### DISCUSSION TITLE
CrySPY version 1.0.0

### DISCUSSION THREAD

**Discussion by Tomoki-YAMASHITA** [Tomoki-YAMASHITA/CrySPY#46] [Announcements]
### Changed
- standard output --> log_cryspy
- standard error --> err_cryspy
- cryspy.out is obsoleted
- COMBO --> optional
- mindist
  + mindist can be omitted in cryspy.in
  + mindist_ea is obsoleted
- moved examples to CrySPY utility
- moved cal_fingerprint program to CrySPY utility
- directory tree (gen_struc, utility, RS, EA)
- CrySPY stops once before going to next selection or next generation

### Added
- excecutable script, cryspy
- auto and manual backup
- mindist_mol_bs and mindist_mol_bs_factor in cryspy.in
- fwpath and fppath in cryspy.in
- new calc_mode: ext

<hr /><em>This discussion was created from the release <a href='https://github.com/Tomoki-YAMASHITA/CrySPY/releases/tag/v1.0.0'>CrySPY version 1.0.0</a>.</em>

**Comment by Tomoki-YAMASHITA** [comment]
CrySPY can be installed by
```
pip3 install csp-cryspy
```

**Comment by yw-fang** [comment]
Hi, Yamashita san, could you provide some introduction about this feature "CrySPY stops once before going to next selection or next generation"? It's a bit difficult to understand what it is to me.

**Reply by Tomoki-YAMASHITA** [reply]
This is not relevant for random search, but for EA, BO, and LAQA.
For example, EA case:

[old version]
1. cryspy run
  - check jobs (finish current generation?)
  - evolutionary algorithm automatically starts (structure generation by EA)

[CrySPY 1.0.0]
1. cryspy run
  - check jobs (finish current generation?)
  - stops
2. cryspy run
  - auto backup
  - evolutionary algorithm automatically starts (structure generation by EA)


This is for safety reasons, because very occasionally EA fails.
I have been writing the document (not uploaded yet).

**Reply by yw-fang** [reply]
Thanks for your explanation! I indeed had such an issue this morning that EA failed in generating new structures. I have never encountered it before. It cost a lot of time to generate the new structures (with many "retry"), and then it was interrupted for some unknown reasons. I managed in generating the required number of new structures by running CrySPY for several times. Is this new feature applicable to solving such an issue?

**Reply by Tomoki-YAMASHITA** [reply]
No, we have not reached a fundamental solution. I'm still not sure when the error occurs and stops. Anyway, I only did an update to prevent the data from getting weird when we come across the issue.
